vault/identity_store_conflicts.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package vault
+
+import (
+	"context"
+
+	log "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/vault/helper/identity"
+)
+
+// ConflictResolver defines the interface for resolving conflicts between
+// entities, groups, and aliases.
+type ConflictResolver interface {
+	ResolveEntities(ctx context.Context, existing, duplicate *identity.Entity) error
+	ResolveGroups(ctx context.Context, existing, duplicate *identity.Group) error
+	ResolveAliases(ctx context.Context, existing, duplicate *identity.Alias) error
+}
+
+// The errorResolver is a ConflictResolver that logs a warning message when a
+// pre-existing Identity artifact is found with the same factors as a new one.
+type errorResolver struct {
+	logger log.Logger
+}
+
+// ResolveEntities logs a warning message when a pre-existing Entity is found
+// and returns a duplicate name error, which should be handled by the caller by
+// putting the system in case-sensitive mode.
+func (r *errorResolver) ResolveEntities(ctx context.Context, existing, duplicate *identity.Entity) error {
+	if existing == nil {
```

Comment:
Should we also `nil` check duplicate before we access `duplicate.Name` below?

Reply:
This is actually kind of  nice because it allows us to search for different-case matches in case-sensitive mode I think. We might want to consider if this is a smart design choice, but I did add a comment in the interface about search flexibility.